magma/interface.py

```diff
@@ -174,23 +196,29 @@ class _DeclareInterface(_Interface):
         interface = Interface()
     """
     def __init__(self, renamed_ports={}, inst=None, defn=None):
-        # Parse the class Interface declaration.
-        names, ports = parse(self.Decl)
+        self.ports = _make_interface_args(self.Decl, renamed_ports, inst, defn)
 
-        args = OrderedDict()
-        for name, port in zip(names, ports):
-            if   inst: ref = InstRef(inst, name)
-            elif defn: ref = DefnRef(defn, name)
-            else:      ref = AnonRef(name)
 
+class _DeclareLazyInterface(_Interface):
+    """_DeclareLazyInterface class"""
+    def __init__(self, renamed_ports={}, inst=None, defn=None):
+        # This interface declaration is only lazy for module definitions (not
+        # instances). If @defn is not supplied, then we use the standard
+        # (non-lazy) interface construction logic.
+        if not defn:
+            self.ports = _make_interface_args(self.Decl, renamed_ports, inst,
+                                              defn)
+            return
+        args = OrderedDict()
+        for name, port in self.io.ports.items():
+            ref = port.name
+            if inst:
+                ref.set_inst(inst)
```

Comment:
I don't see `set_inst` defined in LazyDefnRef, is this a case we need to consider?